pytests/test_datetime.py

```diff
@@ -0,0 +1,166 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+
+from bytewax.dataflow import Dataflow
+from bytewax.execution import run_main
+from bytewax.inputs import ManualInputConfig
+from bytewax.outputs import TestingOutputConfig
+from bytewax.window import TestingClockConfig, TumblingWindowConfig, SystemClockConfig
+
+
+def test_timezone_erase():
+    """
+    Test that we don't erase timezone info when passing a datetime to rust
+    """
+    # In python>=3.9 we have the ZoneInfo package, otherwise we use
+    # a fixed offset as the timezone info
+    try:
+        from zoneinfo import ZoneInfo
+        py39 = True
+        start_at = datetime.now(ZoneInfo("America/Los_Angeles"))
+    except ImportError:
+        py39 = False
+        start_at = datetime.now(tz=timezone(timedelta(hours=-8)))
+
+    wc = TumblingWindowConfig(length=timedelta(seconds=2), start_at=start_at)
+
+    # The dates should always be equivalent
+    assert start_at == wc.start_at
+    # But if we are working with fixed offsets in python<3.9,
+    # we convert everything to UTC, so the tzinfo won't be the same.
+    if py39:
+        assert start_at.tzinfo == wc.start_at.tzinfo
```

Comment:
This would fail due to the choice of always having timezoneaware datetimes internally, and the way I do this conversion (see comments below).
Since I made the ChronoDateTime be `chrono::DateTime<chrono_tz::Tz>`, I can't use a `chrono::FixedOffset`. So when I receive a datetime with a fixed offset from python, i convert it to utc first. This means the `tzinfo` of the object that comes back from rust actually changes, even though it still represents the same date.